src/main/java/modbus/SlaveNodeWithConnection.java

```diff
@@ -0,0 +1,284 @@
+import org.dsa.iot.dslink.node.value.Value;
+import org.dsa.iot.dslink.node.value.ValueType;
+import org.dsa.iot.dslink.util.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.dsa.iot.dslink.util.handler.Handler;
+
+import com.serotonin.modbus4j.ModbusMaster;
+
+public class SlaveNodeWithConnection extends SlaveNode {
+	private static final Logger LOGGER;
+
+	static {
+		LOGGER = LoggerFactory.getLogger(SlaveNode.class);
+	}
+
+	ModbusLink link;
+	long interval;
+
+	SlaveNodeWithConnection(ModbusLink link, final ModbusConnection conn, Node node) {
+		super(conn, node);
+
+		ScheduledThreadPoolExecutor stpe = conn.getDaemonThreadPool();
+		conn.slaves.add(this);
+
+		this.link = link;
+		
+        if(master != null && master.isInitialized()){
+        	statnode.setValue(new Value("Connected"));
+        }
+		stpe.execute(new Runnable() {
+
+			@Override
+			public void run() {
+				conn.checkConnection();
+			}
+
+		});
+
+		this.interval = node.getAttribute("polling interval").getNumber().longValue();
+
+		makeEditAction();
+
+		if (master != null) {
+			makeStopAction();
+		} else {
+			makeStartAction();
+		}
+	}
+
+
+
+	private void makeStartAction() {
+		Action act = new Action(Permission.READ, new Handler<ActionResult>() {
+			public void handle(ActionResult event) {
+				master = getMaster();
+				if (null != master) {
+					conn.checkConnection();
+					node.removeChild("start");
+					makeStopAction();
+				}
+
+			}
+		});
+		Node anode = node.getChild("start");
+		if (anode == null)
+			node.createChild("start").setAction(act).build().setSerializable(false);
+		else
+			anode.setAction(act);
+	}
+
+	private void makeStopAction() {
+		Action act = new Action(Permission.READ, new Handler<ActionResult>() {
+			public void handle(ActionResult event) {
+				if (master != null) {
+					master.destroy();
+					link.masters.remove(master);
+					master = null;
+				}
+				node.removeChild("stop");
+				statnode.setValue(new Value("Stopped"));
+				makeStartAction();
+			}
+		});
+
+		Node anode = node.getChild("stop");
+		if (anode == null)
+			node.createChild("stop").setAction(act).build().setSerializable(false);
+		else
+			anode.setAction(act);
+	}
+
+	private void makeEditAction() {
+		Action act = new Action(Permission.READ, new EditHandler());
+		act.addParameter(new Parameter("name", ValueType.STRING, new Value(node.getName())));
+
+		act.addParameter(
+				new Parameter("transport type", ValueType.makeEnum("TCP", "UDP"), node.getAttribute("transport type")));
+		act.addParameter(new Parameter("host", ValueType.STRING, node.getAttribute("host")));
+		act.addParameter(new Parameter("port", ValueType.NUMBER, node.getAttribute("port")));
+
+		// The device specific parameters
+		act.addParameter(new Parameter("slave id", ValueType.NUMBER, node.getAttribute("slave id")));
+		double defint = node.getAttribute("polling interval").getNumber().doubleValue() / 1000;
```

Comment:
are you sure the polling interval should be a double? I don't know what you're using, but you may consider using an int and use milliseconds.